src/KOKKOS/fftdata_kokkos.h

```diff
@@ -0,0 +1,46 @@
+/* ----------------------------------------------------------------------
+   LAMMPS - Large-scale Atomic/Molecular Massively Parallel Simulator
+   http://lammps.sandia.gov, Sandia National Laboratories
+   Steve Plimpton, sjplimp@sandia.gov
+
+   Copyright (2003) Sandia Corporation.  Under the terms of Contract
+   DE-AC04-94AL85000 with Sandia Corporation, the U.S. Government retains
+   certain rights in this software.  This software is distributed under
+   the GNU General Public License.
+
+   See the README file in the top-level LAMMPS directory.
+------------------------------------------------------------------------- */
+
+#define MAX(A,B) ((A) > (B) ? (A) : (B))
+
+// data types for 2d/3d FFTs
+
+#ifndef FFT_DATA_KOKKOS_H
+#define FFT_DATA_KOKKOS_H
+
+// User-settable FFT precision
+
+// FFT_PRECISION = 1 is single-precision complex (4-byte real, 4-byte imag)
+// FFT_PRECISION = 2 is double-precision complex (8-byte real, 8-byte imag)
+
+#ifdef FFT_SINGLE
```

Comment:
If you pass in `-DFFT_SINGLE` to `Makefile.kokkos_cuda_mpi`, this section of code causes the compile to error out with:

```
../fft3d_kokkos.h(22): error: invalid redeclaration of type name "FFT_SCALAR"
../kokkos_type.h(40): here
```

There are no issues if you pass in `-DFFT_DOUBLE` instead (looks like you can re-`typedef` something at the same scope if you leave it typed as the same thing?), or leave it undefined in `Makefile.kokkos_cuda_mpi` (as it currently stands). 

Not sure how you want to handle it---you could either make the code throw a `#error` if it's defined, or just add to the documentation that you shouldn't define `FFT_SINGLE` or `FFT_DOUBLE` when compiling with Kokkos, or remove this bit of code (or something along those lines).

Reply:
@weinbe2 good point, I'll take a look.